package/kernel/linux/modules/crypto.mk

```diff
@@ -823,6 +835,20 @@ endif
 $(eval $(call KernelPackage,crypto-misc))
 
 
+define KernelPackage/crypto-nhpoly1305
+  TITLE:=Hash function for Adiantum
+  DEPENDS:=+kmod-crypto-hash
+  KCONFIG:= CONFIG_CRYPTO_NHPOLY1305
+  FILES:= \
+        $(LINUX_DIR)/crypto/nhpoly1305.ko \
+        $(LINUX_DIR)/lib/crypto/libpoly1305.ko
```

Comment:
Please do not pack `libpoly1305.ko` into this package. If you need it add a dependency to `+kmod-crypto-lib-poly1305` instead.